### PR TITLE
Locks and better lights in power menu

### PR DESCRIPTION
**Pull request by vigonotion**
- Add locks to power menu

- Only show range slider for lights when lights actually support brightness (via supported_features)

photo_2020-10-24_16-32-54

### CODE REVIEW

**Review by dshokouhi — APPROVED** [review]
Looks good to me

File: app/src/main/java/io/homeassistant/companion/android/controls/LightControl.kt [hunk 1]
```diff
@@ -40,22 +42,31 @@ class LightControl {
...
-                        100f,
-                        (entity.attributes["brightness"] as? Number)
-                            ?.toFloat()
-                            ?.div(255f)
-                            ?.times(100) ?: 0f,
-                        1f,
-                        "%.0f%%"
-                    )
-                )
+                    if ((entity.attributes["supported_features"] as Int) and 1 == 1)
```
Comment (JBassett) [inline app/src/main/java/io/homeassistant/companion/android/controls/LightControl.kt:45]: This some debugging logic I'm guessing?

**Review by JBassett — CHANGES_REQUESTED** [review]

File: app/src/main/java/io/homeassistant/companion/android/controls/LightControl.kt (hunk 1 above)
Comment (vigonotion) [reply app/src/main/java/io/homeassistant/companion/android/controls/LightControl.kt:45]: That's a bitwise operation. (Supported features AND 1) should be equal to one, it just looks a bit strange

Comment (JBassett) [reply app/src/main/java/io/homeassistant/companion/android/controls/LightControl.kt:45]: Can we add a constant so we know 1 the first bit is the brightness support?

Comment (vigonotion) [reply app/src/main/java/io/homeassistant/companion/android/controls/LightControl.kt]: yes good idea, I added it now

**Review by JBassett — APPROVED** [review]

**Pull request merged**